File: post-sample-server/src/main/java/com/hunseong/postsample/repository/CommentRepository.java
package com.hunseong.postsample.repository;

import com.hunseong.postsample.entity.Comment;
import org.springframework.data.jpa.repository.JpaRepository;

/**
 * Created by Hunseong on 2022/06/14
 */
public interface CommentRepository extends JpaRepository<Comment, Long> {
}

File: post-sample/app/src/main/java/hunseong/com/post_practice/data/repository/CommentRepository.kt
package hunseong.com.post_practice.data.repository

import hunseong.com.post_practice.data.api.CommentApiService
import hunseong.com.post_practice.data.entity.Comment
import io.reactivex.rxjava3.android.schedulers.AndroidSchedulers
import io.reactivex.rxjava3.core.Single
import io.reactivex.rxjava3.schedulers.Schedulers
import javax.inject.Inject

class CommentRepository @Inject constructor(
    private val commentApiService: CommentApiService,
) : Repository {
    fun getComments(postId: Long): Single<List<Comment>> {
        return commentApiService.getComments(postId)
            .observeOn(AndroidSchedulers.mainThread())
            .subscribeOn(Schedulers.io())
    }
}